lib/chef-dk/command/generator_commands/cookbook.rb

```diff
@@ -191,6 +191,11 @@ def read_and_validate_params
             @params_valid = false
           end
 
+          if /-/ =~ @cookbook_name_or_path
+            err("Hyphens are not allowed in cookbook names. Please specify a cookbook name with hyphens.")
```

Comment:
Also, the intermediate path could have a hyphen in it. We should use some path separator shenanigans to determine if the cookbook name is the thing with the hyphen in it